angular/playground/projects/ng-entity-service/src/lib/ng-entity.service.ts

```diff
@@ -23,44 +22,58 @@ import {
 import { isID } from './helpers';
 import { errorAction, successAction } from './action-factory';
 
-export const mapResponse = (config: HttpConfig) =>
-  map(res => ((config || {}).mapResponseFn ? config.mapResponseFn(res) : res));
+export const mapResponse = <T>(config?: HttpConfig<T>) =>
+  map(res => (config && !!config.mapResponseFn ? config.mapResponseFn(res) : res));
 
 export class NgEntityService<S extends EntityState = any> extends EntityService<S> {
-  baseUrl: string;
+  baseUrl?: string;
   loader: NgEntityServiceLoader;
 
-  private http: HttpClient;
-  private notifier: NgEntityServiceNotifier;
-  private globalConfig: NgEntityServiceGlobalConfig = {};
-  private mergedConfig: NgEntityServiceParams & NgEntityServiceGlobalConfig;
+  private readonly http: HttpClient;
+  private readonly notifier: NgEntityServiceNotifier;
+  private readonly mergedConfig: NgEntityServiceParams & NgEntityServiceGlobalConfig;
+  private readonly httpMethodMap:
+    | Partial<{
+        GET: HttpMethod;
+        POST: HttpMethod;
+        PATCH: HttpMethod;
+        PUT: HttpMethod;
+        DELETE: HttpMethod;
+      }>
+    | undefined;
 
-  private dispatchSuccess: (action: Partial<EntityServiceAction>) => void;
-  private dispatchError: (action: Partial<EntityServiceAction>) => void;
+  private readonly dispatchSuccess: (action: Partial<EntityServiceAction>) => void;
+  private readonly dispatchError: (action: Partial<EntityServiceAction>) => void;
 
-  constructor(protected store: EntityStore<S>, private config: NgEntityServiceParams = {}) {
+  constructor(protected readonly store: EntityStore<S>, readonly config: NgEntityServiceParams = {}) {
     super();
     this.http = inject(HttpClient);
     this.loader = inject(NgEntityServiceLoader);
     this.notifier = inject(NgEntityServiceNotifier);
-    this.globalConfig = inject(NG_ENTITY_SERVICE_CONFIG);
+    const globalConfig = inject(NG_ENTITY_SERVICE_CONFIG);
 
-    this.mergedConfig = mergeDeep(defaultConfig, this.globalConfig, config);
+    this.mergedConfig = mergeDeep(defaultConfig, globalConfig, config);
```

Comment:
We should have one config based on this order:
- default
- global
- decorator
- inline

Reply:
this is great. It'll clean up the config confusion 👍